### PR TITLE
Refactored `anyio.run()` and `TestRunner` on asyncio to use `asyncio.Runner`

### LINKED ISSUE #305: starting a blocking portal permenantly sets the process global event loop policy in a thread unsafe way (when a policy is passed)
also applies to `anyio.run` but that's mainly run from the main thread with `sys.exit(anyio.run(amain))` so is safer

Note: this issue was previously about the default behavior of opting into the uvloop policy

### PR DESCRIPTION
**Pull request by agronholm**
Fixes #305.

### CODE REVIEW

File: src/anyio/_backends/_asyncio.py [hunk 1]
```diff
@@ -1752,65 +1881,65 @@ async def _call_in_runner_task(
...
-                self._loop.run_until_complete(self._loop.shutdown_default_executor())
-            elif isinstance(self._loop, asyncio.BaseEventLoop) and hasattr(
-                self._loop, "_default_executor"
-            ):
-                # asyncio in Python < 3.9
-                self._loop.run_until_complete(_shutdown_default_executor(self._loop))
-        finally:
-            asyncio.set_event_loop(None)
-            self._loop.close()
+    # def close(self) -> None:
```
Comment (graingert) [inline src/anyio/_backends/_asyncio.py:1893]: Big block of commented code?

Comment (agronholm) [reply src/anyio/_backends/_asyncio.py:1893]: Oops

**Comment by agronholm** [comment]
I've documented the change to `anyio.run()` now. Is this okay?

**Comment by graingert** [comment]
Yeah looks good. I'm still grumpy about the ki handling in asyncio compared to trio

**Review by graingert — APPROVED** [review]

**Pull request merged**